src/components/ConnectionStatus.tsx

```diff
@@ -0,0 +1,22 @@
+import React, { useCallback, useEffect, useState } from 'react'
+import { Notification } from './Notification'
+
+export const ConnectionStatus: React.FC = () => {
+  const [online, setOnline] = useState(true)
+
+  const handleConnectionChange = useCallback((e: Event) => {
+    setOnline(e.type === 'online')
+  }, [])
+
+  useEffect(() => {
+    window.addEventListener('online', handleConnectionChange)
+    window.addEventListener('offline', handleConnectionChange)
+
+    return (): void => {
+      window.removeEventListener('online', handleConnectionChange)
+      window.removeEventListener('offline', handleConnectionChange)
+    }
+  }, [handleConnectionChange])
+
+  return online ? null : <Notification type="warn" message="Your internet connection is down." appendMessage={false} />
```

Comment:
I think this message is maybe assuming too much.

What about using the same message as Cowswap/Uniswap? https://github.com/gnosis/cowswap/blob/ac962fb00d457bc2c4f59432d7d6d7741443dfea/src/components/Header/ChainConnectivityWarning.tsx#L62

```suggestion
  return online ? null : <Notification type="warn" message="You may have lost your network connection." appendMessage={false} />
```